File: SP-Client/src/app/_models/orderItemBatch.ts
import { ItemBatch } from "./itemBatch";

export class OrderItemBatch {
  id: number;
  orderId: number;
  itemBatchId: number;
  orderItemBatchCode: string;
  itemBatchCode: string;
  name: string;
  itemCount: number;
  itemPrice: number;
  shopOwnerDiscountRate: number;
  netPrice: number;
  totalNetPrice: number;
  orderedDate: Date;

  constructor(orderItemBatch) {
    this.id = orderItemBatch.id;
    this.orderId = orderItemBatch.orderId;
    this.itemBatchId = orderItemBatch.itemBatchId;
    this.orderItemBatchCode = orderItemBatch.orderItemBatchCode;
    this.name = orderItemBatch.name;
    this.itemCount = orderItemBatch.itemCount;
    this.itemPrice = orderItemBatch.itemPrice;
    this.shopOwnerDiscountRate = orderItemBatch.shopOwnerDiscountRate;
    this.netPrice = orderItemBatch.netPrice;
    this.totalNetPrice = orderItemBatch.totalNetPrice;
    this.orderedDate = orderItemBatch.orderedDate;
  }
}

File: SP-API/Domain/OrderItemBatch.cs
namespace Domain
{
    public class OrderItemBatch
    {
        public int Id { get; set; }
        public int OrderId { get; set; }
        public int ItemBatchId { get; set; }
        public string OrderItemBatchCode { get; set; }
        public string Name { get; set; }
        public int ItemCount { get; set; }
        public decimal ItemPrice { get; set; }
        public decimal ShopOwnerDiscountRate { get; set; }
        public decimal NetPrice { get; set; }
        public decimal TotalNetPrice { get; set; }
        public virtual ItemBatch ItemBatch { get; set; }
        public virtual Order Order { get; set; }
    }
}